sympy/core/numbers.py

```diff
@@ -1754,18 +1764,23 @@ def __div__(self, other):
                     return S.ComplexInfinity
                 else:
                     return Rational(self.p, self.q*other.p, igcd(self.p, other.p))
+            elif isinstance(other, DecimalRational):
```

Comment:
Why do all the arithmetic operations on `Rational` need to know about `DecimalRational`?

Reply:
If you change `Rational` in [this](https://github.com/meganly/sympy/blob/17648_DecimalRationalclass/sympy/core/numbers.py#L1705) line to `self.__class__` you run into a problem with `S.Half`.  The only workarounds I saw were to write new methods for `S.Half` or write methods for `DecimalRational`.  I chose the latter so that the conventions`DecimalRational + Rational = DecimalRational` and `Rational + DecimalRational = DecimalRational` would be consistent.  Also to check for perfect squares so that things like `sqrt(.25)` could be evaluated to `sqrt(.5)`, but not rewrite things like `sqrt(.8)` as `2*sqrt(5)/5`.

Reply:
I refactored the code to use `op_priority` and a `_classof` method so the arithmetic operations on `Rational` do not need to know about `DecimalRational`